### PR TITLE
Extend the cucumber spec around references

### PR DESCRIPTION
**Pull request by benilovj**
### Context

To improve the documentation around the business rules.

### Changes proposed in this pull request

- Add a bit more structure to the free-text bit of the spec
- The two existing scenarios are generalised using a scenario table. This change
also adds coverage for some cases that weren't covered by the original scenarios.

### CODE REVIEW

File: features/submitting_references.feature [hunk 1]
```diff
@@ -2,33 +2,31 @@
 Feature: references
   Candidates need two references as part of their application. When filling in the application form, they provide two sets of contact details and DfE then prompts the referees for feedback once the candidate has submitted the form.
 
+  Swapping out referees
+  =====================
   If a referee don't provide a reference within a certain period of time, a candidate is allowed to swap out that referee for another one. Candidates can't remove or swap out references once they've been submitted by the referees.
```
Comment (fofr) [inline features/submitting_references.feature]:
```suggestion
  If a referee doesn't provide a reference within a certain period of time, a candidate is allowed to swap out that referee for another one. Candidates can't remove or swap out references once they've been submitted by the referees.
```

File: features/step_definitions/application_steps.rb [hunk 2]
```diff
@@ -31,10 +31,30 @@
...
+    And the candidate submits the application
+  }
+end
+
 When(/^the candidate submits the application$/) do
   SubmitApplication.new(@application_choice.application_form).call
 end
 
+When('{int} referees complete the references') do |number_of_complete_references|
+  references = @application_choice.application_form.reload.references[0...number_of_complete_references.to_i]
```
Comment (tijmenb) [inline features/step_definitions/application_steps.rb]:
```suggestion
  references = @application_choice.application_form.reload.references.first(number_of_complete_references)
```

Comment (tijmenb) [reply features/step_definitions/application_steps.rb]: (feel free to ignore)

Comment (benilovj) [reply features/step_definitions/application_steps.rb]: Yeah nice 👍, added

**Pull request merged**